packages/driver/src/cy/multi-domain/index.ts

```diff
@@ -141,9 +142,13 @@ export function addCommands (Commands, Cypress: Cypress.Cypress, cy: Cypress.cy,
           // is common if there are only assertions enqueued in the SD.
           if (!commandsManager.hasCommands && !done) {
             cleanup()
-          }
+            // This handles when a subject is returned synchronously
+            const deserializedSubject = deserialize(subject)
 
-          resolve()
+            resolve(deserializedSubject)
```

Comment:
Here is where we resolve anything `returned` from the switchToDomain function. I'm not sure when it makes sense to synchronously return something from `switchToDomain`, but i'm matching the functionality of the `then` function blocks.